osquery/tables/networking/linux/iptables.cpp

```diff
@@ -71,6 +72,40 @@ void parseIpEntry(const ipt_ip *ip, Row &r) {
   r["outiface_mask"] = TEXT(outiface_mask);
 }
 
+void parseEntryMatch(const struct ipt_entry* en, Row& r) {
+  // Get rule port details from the xt_entry_match object
+
+  // m will never be NULL
+  auto m = (struct xt_entry_match*)en->elems;
+
+  if (en->ip.proto == IPPROTO_TCP) {
+    auto m_data = (struct ipt_tcp*)m->data;
+    r["src_port"] = (m_data != nullptr)
+                        ? std::to_string(m_data->spts[0]) + ':' +
+                              std::to_string(m_data->spts[1])
+                        : "-1";
```

Comment:
Right, to @obelisk's point later in the code review about possible values for ports: for `TEXT` types, it's OK to have `NULL` fields. So you can choose to leave this blank, instead of '-1'.